### PR TITLE
[Authorization] Converge authz of ns policies from super-user to tenant-administrator

### PR DESCRIPTION
**Pull request by yuruguo**
### Motivation
Currently, only the `super-user` has the authority to operate part of the namespace policies. In fact, `tenant-administrator` of this namespace should also be able to operate.
The namespace policies involved includes:
- publish-rate
- topic-dispatch-rate
- subscription-dispatch-rate
- subscribe-rate
- replicator-dispatch-rate
- inactive-topic
- delayed-delivery
- max-subscriptions-per-topic

### Documentation  
- [x] `no-need-doc`

### CODE REVIEW

**Comment by gaoran10** [comment]
If all consume roles could modify the namespace level subscription-related policies, do it may cause some confusion?

**Comment by yuruguo** [comment]
> If all consume roles could modify the namespace level subscription-related policies, do it may cause some confusion?

In fact, this problem still exists if the tenant has multiple `admin-roles`, maybe we need to discuss it separately. 
https://github.com/apache/pulsar/blob/47e4f38f2ed10f9b9492ebeb3e1424da18ab3655/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTenants.java#L61-L63
But in terms of the operation permission for subscription-related policies, it should be more reasonable to converge to a role that can `consume` namespace.

**Comment by yuruguo** [comment]
@eolivelli @codelipenghui @hangc0276 @gaoran10 Would you help review this PR? thx!

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
I think this PR introduces breaking changes for any existing third party implementations of the `AuthorizationProvider`.

@yuruguo - is your proposal to make it possible for a non admin user to have sufficient permission to update these namespace policies? As currently written, I think these operations fall within the role of the tenant admin (although I'm not sure we have this defined explicitly somewhere).

(Requesting changes to prevent this from being merged for now.)

**Comment by yuruguo** [comment]
/pulsarbot run-failure-checks

> I think this PR introduces breaking changes for any existing third party implementations of the `AuthorizationProvider`.
> 
> @yuruguo - is your proposal to make it possible for a non admin user to have sufficient permission to update these namespace policies? As currently written, I think these operations fall within the role of the tenant admin (although I'm not sure we have this defined explicitly somewhere).
> 
> (Requesting changes to prevent this from being merged for now.)

I have updated [Motivation](https://github.com/apache/pulsar/pull/13157#issue-1072782305) and Code, PTAL, thx!

**Comment by yuruguo** [comment]
> I think this PR introduces breaking changes for any existing third party implementations of the `AuthorizationProvider`.
> 
> @yuruguo - is your proposal to make it possible for a non admin user to have sufficient permission to update these namespace policies? As currently written, I think these operations fall within the role of the tenant admin (although I'm not sure we have this defined explicitly somewhere).
> 
> (Requesting changes to prevent this from being merged for now.)

I have updated [Motivation](https://github.com/apache/pulsar/pull/13157#issue-1072782305) and Code, PTAL, thx!

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 1]
```diff
@@ -122,7 +122,7 @@
     @Path("/{property}/{cluster}/{namespace}/destinations")
     @ApiOperation(hidden = true, value = "Get the list of all the topics under a certain namespace.",
             response = String.class, responseContainer = "Set")
-    @ApiResponses(value = {@ApiResponse(code = 403, message = "Don't have admin permission"),
+    @ApiResponses(value = {@ApiResponse(code = 403, message = "Don't have admin or consume permission on namespace"),
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:125]: Maybe we need to modify the message, such as "Don't have admin or operate policies permission on the namespace".

**Review by gaoran10 — COMMENTED** [review]
Overlook good to me.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 2]
```diff
@@ -1357,7 +1357,7 @@ protected void internalRemoveReplicatorDispatchRate() {
     }
 
     protected void internalSetReplicatorDispatchRate(DispatchRateImpl dispatchRate) {
-        validateSuperUserAccess();
+        validateNamespacePolicyOperation(namespaceName, PolicyName.RATE, PolicyOperation.WRITE);
```
Comment (michaeljmarshall) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1360]:
```suggestion
        validateNamespacePolicyOperation(namespaceName, PolicyName.REPLICATION_RATE, PolicyOperation.WRITE);
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 3]
```diff
@@ -1341,7 +1341,7 @@ protected SubscribeRate internalGetSubscribeRate() {
     }
 
     protected void internalRemoveReplicatorDispatchRate() {
-        validateSuperUserAccess();
+        validateNamespacePolicyOperation(namespaceName, PolicyName.RATE, PolicyOperation.WRITE);
```
Comment (michaeljmarshall) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1344]:
```suggestion
        validateNamespacePolicyOperation(namespaceName, PolicyName.REPLICATION_RATE, PolicyOperation.WRITE);
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java (hunk 1 above)
Comment (michaeljmarshall) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:125]: @gaoran10 good point. The risk here is that custom implementations won't necessarily align with needing admin privileges. I think making it an "or" like you proposed sounds good to me.

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
I think this is headed in the right direction. I think there are just a couple of changes we need before this can get merged.

It is very similar to a PR I opened a while ago that got stalled: https://github.com/apache/pulsar/pull/11872. I will follow up on that PR soon.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java (hunk 1 above)
Comment (yuruguo) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:125]: So “ Don't have operate policies permission on the namespace” is better? @michaeljmarshall @gaoran10

**Comment by yuruguo** [comment]
@merlimat @eolivelli @codelipenghui @hangc0276 ptal

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java (hunk 1 above)
Comment (yuruguo) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java]: @gaoran10 @michaeljmarshall Thanks for you suggestion :)
This is a correction of known inaccurate annonations, not limited to `namespace polocies` operation, including: `getTopics`, `clearNamespaceBundleBacklog`, `unsubscribeNamespace`, etc. so perhaps `Don't have admin or operate permission on the namespace` is more general, what do you think?

**Review by eolivelli — APPROVED** [review]
+1

**Comment by yuruguo** [comment]
@codelipenghui @freeznet PTAL, thx!

**Review by michaeljmarshall — APPROVED** [review]
LGTM. Thanks for taking care of all of my comments @yuruguo!

**Review by 315157973 — APPROVED** [review]
Nice work!

**Comment by michaeljmarshall** [comment]
@yuruguo - would you mind renaming this PR's title before we merge it? That way the commit message aligns with the changes. Thanks!

**Comment by hsaputra** [comment]
This is another breaking changes that impact the  implementations of the AuthorizationProvider =(

Could we be more gentle and considerate when making these kind of "improvements " ?

**Comment by eolivelli** [comment]
@hsaputra in this case we have not broken compatibility.
What do you mean?

**Comment by hsaputra** [comment]
Hi @eolivelli - sorry, I did not mean broken compatibility, but more of may add potential of breaking changes for external implementation of AuthorizationProvider to handle new operations need to be handled.

**Comment by hsaputra** [comment]
But agree I may be bit over-worried on this change particular changes.
Just bit worry since we have to update implementation of our plugins from last time we sync our internal repo =(

**Comment by michaeljmarshall** [comment]
Hi @hsaputra - you're right that it would impact external implementations. There is already a PR to revert this commit. There is ongoing discussion there. https://github.com/apache/pulsar/pull/13463

**Comment by hsaputra** [comment]
Thanks for the follow up, @michaeljmarshall !

**Pull request merged**